symbolic-base/src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -168,15 +168,12 @@ instance (Symbolic c, KnownNat n) => BoolType (ByteString n c) where
     false = fromConstant (0 :: Natural)
     true = not false
 
-    not (ByteString bits) =  ByteString $ fromCircuitF bits solve
-        where
-            solve :: MonadCircuit i (BaseField c) m => Vector n i -> m (Vector n i)
-            solve xv = do
-                let xs = V.fromVector xv
-                ys <-  for xs $ \i -> newAssigned (\p -> one - p i)
-                return $ V.unsafeToVector ys
+    not (ByteString bits) = ByteString $ fromCircuitF bits $ \xv -> do
+        let xs = V.fromVector xv
+        ys <- for xs $ \i -> newAssigned (\p -> one - p i)
+        return $ V.unsafeToVector ys
```

Comment:
```suggestion
    not (ByteString bits) = ByteString $ fromCircuitF bits $ mapM (\i -> newAssigned (\p -> one - p i))
```
Vectors are Traversable, would this work?

Reply:
Yes, this would, thank you!